### ISSUE TITLE
Improving the File > Open dialogue

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#4257]
1. First a bug.  We had situations where we were importing multiple (identical) sheets from Excel.
The first sheet worked OK.  When you return to the dialogue the Excel options are not visible.  They re-opened when you returned to the file and opened it again.
2.Try this with Open From Library > Instat Datasets and choose the Introductory guide.  Open the survey.  Then use the short-cut of going back to this dialogue. 
3.  A second issue we faced is when there are multiple missing value codes in a numeric column.  We then currently import as text i.e. as type (c).  If you then change to numeric you get very strange results, because the numbers are now value labels.  You can change the odd codes all to missing (I think) , but we need other options as well.  One easy option is to allow multiple missing values in the file.  In examples last week they were various combinations of <blank>, ".", "NA" and "*".  Could <blank> always be missing?  Could we choose more than one of these.  What other options could be allowed on input. 

4. We will also never cover everything, so we need to have a health warning that after importing if some columns are given as (c) when they should be numeric then something has probably gone wrong in the import.  

5. Once this is fixed, could we consider 2 possible enhancements.  The reason for going back to the same Excel file is often to read another sheet.  Could we have an **option to read multiple sheets** at the same time.  The restriction is that they must have the same options.  They go into separate data frames in R-Instat.

6. Similarly I had the situation recently of reading multiple csv files.  Again it would be great if they could be read at one go.

**Comment by rdstern** [comment]
Some of the above have now been implemented.  We have been moving ahead.

I now have a (very?) small request, which is to be able to import files with multiple missing value codes.  I think this should be easy with a small change in the code to generate the command.  Currently what is generated in the command when I type '9999','9988' into the missing box is , `na="c('9999','9988')"`
 which doesn't work.  Instead it should be `na = c('9999', '9988')`.  
If, as usual, it is a single missing value code, then it produces `na="9999"`, so it is just adding the quotes round what is in the field.
Ideally I would like to be able type just `9999,9988` and it would detect the comma and do the rest for me.
Am I right that this is easy?  I ran into this issue with the first example I tried to import to put into the Help - I can - of course - use many other examples, but feel that using this exercise to "tighten" our dialogues would be useful.

**Comment by Patowhiz** [comment]
Yes that's doable. What about if the missing value placeholder contains a comma? Do we have such scenarios? @dannyparsons what's your suggestion on how to simplify this from a user's perspective. I can quickly implement this @rdstern way then in future this gets simplified further

**Comment by rdstern** [comment]
Sounds good.  I suggest having a comma in the number, or text, to be set to missing is rare enough that we detect multiple missing that way.  I will check with @dannyparsons

**Comment by Patowhiz** [comment]
Okay waiting for @dannyparsons opinion

**Comment by rdstern** [comment]
I checked with @volloholic who had a good idea!  (He does, just occasionally!)  This was to have a checkbox below the missing value field that says Multiple missing values.  Default is unchecked.  If checked, then it opens a field to type into for a second missing value.  Then this is combined with the first missing value code.  It there are more than 2 - which we think is rare, then this extra box can take multiple values separated by commas.  The first box just has a single value, as now, so is taken complete - whatever it contains.

**Comment by Patowhiz** [comment]
Yes that's doable. I will now go ahead and implement this

**Comment by Patowhiz** [comment]
I think once PR #6362 is merged then all the items outlined here in regards to the dialog would have been fixed.

**Comment by lilyclements** [comment]
1-3 are now fixed
4-5 have been discussed in another issue (#7289)
